ReSharper.FSharp/src/FSharp.Psi.Services/src/Generate/GenerateProvider.fs

```diff
@@ -230,38 +233,34 @@ type FSharpOverridingMembersBuilder() =
                         shiftWithWhitespaceBefore -diff f)
         | _ -> ()
 
-        // members might need deindenting after the representation start was moved to it's own line
-        typeDecl.TypeMembers
+        let deindent indentationTarget (nodes: seq<ITreeNode>) =
+            nodes
+            |> Seq.iter (fun m ->
+                let diff = m.Indent - indentationTarget
+                if diff > 0  && isFirstMeaningfulNodeOnLine m then shiftWithWhitespaceBefore -diff m)
+
+        // members etc. might need indenting or deindenting after the representation start was moved to it's own line
+        let reprNodes =
+            typeRepr.Children()
+            |> Seq.filter
+                   (fun c -> c :? ICommentNode || c :? ITypeBodyMemberDeclaration || c :? IRecordFieldDeclarationList)
```

Comment:
Does it need to be specific about the node types? Maybe it could skip `{`/`class`/`struct` and move any nodes until `}`/`end`?